spring-boot-admin-starter-client/src/main/java/de/codecentric/boot/admin/config/AdminClientProperties.java

```diff
@@ -134,10 +134,7 @@ public String getServiceUrl() {
 		}
 
 		if (preferIp) {
-			InetAddress address = server.getAddress();
-			if (address == null) {
-				address = getHostAddress();
-			}
+			InetAddress address = getHostAddress();
```

Comment:
Move the local variable address out of this block and get rid of the getHostAddress() call in line 138.